### ISSUE TITLE
Clarify where to use C and C++

### ISSUE THREAD

**Issue by jdemel** [gnuradio/volk#281]
This is an issue I stumbled upon for some time. While the VOLK 'infrastructure' does use C++ and Python in some places, the actual library is written in C. I suggest to clarify this structure in the README for new users.

**Comment by michaelld** [comment]
+1 from me for someone to do this. Make it so, I say!

**Comment by jdemel** [comment]
Where should this be documented? Maybe in the project README?

**Comment by michaelld** [comment]
Yes I'd say the top-level project README is the best place. Shouldn't be a long section I'd think!

**Comment by jdemel** [comment]
Our current structure looks like this 	
- apps 
     - Python and C++
- cmake 	
     - CMake stuff obviously
- docs
     - Doxygen
- gen
     - Python: Build system
- include/volk
     - Headers to include VOLK in C and C++
     - C++ specific headers for `volk::vector`
- kernels 	
     - C kernels, should only be C but C++ is present as well.
- lib
     - C and C++ for tests. 	
- python
     - Python: volk_modtool
- scripts
- tmpl
     - Templates: Build system
- tools
     - Tools for releases etc.

Basically, I'd suggest we stick with C in kernels. As well as everything that is part of libvolk.so. Everything else are tools that may be C/C++/Python/Bash.
Also, scripts and tools may be merged into one folder.

**Comment by michaelld** [comment]
Trying to clean out old stuff & came upon this! Still relevant, I think!

I'd say if you combine scripts and tools as noted, then your list here is good to go. I'd say to add it to the top-level README. It is handy!

**Comment by jdemel** [comment]
Yeah. I'll put together a new PR.
@michaelld Are you in favor of merging scripts and tools into on dir?

**Comment by michaelld** [comment]
@jdemel yes! they are just related scripts & tools to do stuff that Volk needs but unrelated to the build. 1 directory is easier than 2 here!
